### PR TITLE
chore: remove installation of redundant detox-cli in bitrise

### PR DESCRIPTION
**Pull request by legobeat**
The package `detox-cli` is just acting as a shim for `detox` and isn't actually needed: https://github.com/wix/Detox/blob/master/detox-cli/cli.js 

- ci: No global installation of `detox-cli` package
- Remove redundant `setup:detox-e2e` package script
- ci: Skip `applesimutils` installation for bitrise android e2e
- ci: Properly pass `HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK` to brew
- devDeps: bump `detox` to latest

### CODE REVIEW

**Comment by cortisiko** [comment]
@legobeat, I recently merged a [PR](https://github.com/MetaMask/metamask-mobile/pull/10553) that removed detox from the setup script. Having detox cli as a dev dependency makes sense. Perhaps resolve the conflict with main on this PR and we can aim to get this merge

**Comment by cortisiko** [comment]
> @cortisiko Rebased.
> 
> Might I ask, why was the approach (installing it globally in bitrise, and leaving it as exercise for the dev elsewhere) in #10553 preferred over this PR (install it through devDependencies in every environment, like any other devdep)?


@legobeat  ah, hmmm, so I wasn't aware you had opened this PR, so it wasn't a matter of preference. Installing Detox as a dev dependency makes sense to me. Before merging this PR, Can we kick off an e2e test build on bitrise for good measure?

**Comment by cortisiko** [comment]
oooooo @legobeat the e2e tests are not going to run on a fork. Can you open the PR the mobile repo?

**Comment by cortisiko** [comment]
@legobeat seems like the build failed again. to confirm `detox-cli-devdeps` is the PR you are working on?

> It seems that the SonarCloud CI job is still broken and needs addressing but can be bypassed? Any concerns wrt that for this PR?

I wouldn't say Sonar is broken, perhaps it is a limitation on running against forks? Not sure. But in this instance, we can ignore it for now.

**Review by gambinish — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -20,7 +20,6 @@
     "setup": "yarn clean && node scripts/setup.mjs",
     "setup:node": "yarn allow-scripts && yarn patch-package",
     "setup:e2e": "cd wdio && yarn install",
-    "setup:detox-e2e": "./scripts/detox-setup.sh",
```
Comment (NicolasMassart) [inline package.json:23]: S: should you also delete the ./scripts/detox-setup.sh file?

Comment (legobeat) [reply package.json:23]: Should it be done by someone else? :P

Either way, PR updated with removal!

**Review by Cal-L — DISMISSED** [review]
LGTM

**Review by Cal-L — COMMENTED** [review]
Ah looks like Bitrise is failing - https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/9c5418c8-da02-491e-9311-be099ec9e853

File: bitrise.yml [hunk 2]
```diff
@@ -833,7 +834,7 @@ workflows:
           inputs:
             - content: |-
                 #!/usr/bin/env bash
-                brew install ccache with HOMEBREW_NO_DEPENDENTS_CHECK=1
+                HOMEBREW_NO_DEPENDENTS_CHECK=1 brew install ccache
```
Comment (legobeat) [inline bitrise.yml:837]: `HOMEBREW_NO_DEPENDENTS_CHECK` is expected as an environment variable.

Fixes Bitrise error when installing ccache: `Error: No formulae or casks found for with homebrew_no_dependents_check=1.`

- Originally introduced in #9611

**Comment by legobeat** [comment]
@Cal-L: Passing bitrise run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/5d58b4ec-72d9-4e3e-b1e9-020db0639848

`detox-cli` is just a shim for the `detox` binary provided by package depenedncy `detox` so it can actually be fully removed.

**Review by cortisiko — APPROVED** [review]
LGTM!

**Pull request merged**